src/mw-ui-components/directives/mw_tab_bar.js

```diff
@@ -1,61 +1,118 @@
 angular.module('mwUI.UiComponents')
-  //TODO rename
-  /**
-   * @example ```html
-   * <!-- change callback example -->
-   * <div mw-tabs active-pane-number="myCtrl.activePane" tab-changed="myCtrl.tabChanged">
-      <div mw-tabs-pane="{{'mytitle'| i18n}}">
-      Tab 1
-      </div>
-      <div mw-tabs-pane="{{'mytitle_2'| i18n}}">
-      Tab 2
-      </div>
-    </div>
-   * ```
-   */
-  .directive('mwTabs', function () {
+//TODO rename
+/**
+ * @example ```html
+ * <!-- change callback example -->
+ * <div mw-tabs active-pane-number="myCtrl.activePane" tab-changed="myCtrl.tabChanged">
+ <div mw-tabs-pane="{{'mytitle'| i18n}}">
+ Tab 1
+ </div>
+ <div mw-tabs-pane="{{'mytitle_2'| i18n}}">
+ Tab 2
+ </div>
+ </div>
+ * ```
+ */
+  .directive('mwTabs', function ($rootScope) {
     return {
       transclude: true,
       scope: {
-        justified: '=',
-        activePaneNumber: '=',
-        tabChanged: '='
+        justified: '=?',
+        activePaneNumber: '=?',
+        activePaneId: '@?',
+        tabChanged: '=?'
       },
       templateUrl: 'uikit/mw-ui-components/directives/templates/mw_tab_bar.html',
       controller: function ($scope) {
-        var panes = $scope.panes = [];
-        $scope.select = function (pane) {
-          angular.forEach(panes, function (p) {
-            p.selected = false;
+        var panes = $scope.panes = [],
+          activeNumber;
+
+        var setInitialSelection = function () {
+          activeNumber = null;
+          if (angular.isUndefined($scope.activePaneNumber) && angular.isUndefined($scope.activePaneId)) {
+            $scope.select(panes[0]);
```

Comment:
if the panes array is not defined at the beginning and activePaneNumber and activePaneId is not defined this will crash.

Reply:
No it won't crash because panes is initialised as an array
```
var panes = $scope.panes = [],
```